File: unimonpark/unimonpark-frontend/src/types/usuario.ts
export interface Usuario {
    idUsuario: number;
    nombres: string;
    apellidos: string;
    documento: string;
    correo: string;
    telefono: string | null;
    nombreUsuario: string | null;
    idRol: number;
    activo: boolean;
    fechaCreacion?: string;
    fechaActualizacion?: string | null;
    valorMatricula: number | null;
    valorSalario: number | null;
}

export interface UsuarioPayload {
    nombres: string;
    apellidos: string;
    documento: string;
    correo: string;
    telefono: string;
    nombreUsuario?: string;
    contrasena?: string;
    idRol: number;
    activo: boolean;
    valorMatricula: number | null;
    valorSalario: number | null;
}

File: unimonpark/src/main/java/co/edu/unimonserrate/unimonpark/entity/Usuario.java
package co.edu.unimonserrate.unimonpark.entity;

import java.math.BigDecimal;
import java.time.LocalDateTime;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Table(name = "usuarios")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class Usuario {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long idUsuario;

    @Column(nullable = false)
    private String nombres;

    @Column(nullable = false)
    private String apellidos;

    @Column(nullable = false, unique = true)
    private String documento;

    @Column(nullable = false, unique = true)
    private String correo;

    private String telefono;

    @Column(name = "nombre_usuario", nullable = false, unique = true)
    private String nombreUsuario;

    @Column(name = "contrasena_hash", nullable = false)
    private String contrasenaHash;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "id_rol", nullable = false)
    private Rol rol;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "id_perfil_usuario_tarifa")
    private PerfilUsuarioTarifa perfilUsuarioTarifa;

    @Column(name = "voluntario_centro_obrero", nullable = false)
    private Boolean voluntarioCentroObrero = false;

    @Column(nullable = false)
    private Boolean activo;

    @Column(name = "fecha_creacion", nullable = false)
    private LocalDateTime fechaCreacion;

    @Column(name = "fecha_actualizacion")
    private LocalDateTime fechaActualizacion;

    @Column(name = "valor_matricula", precision = 12, scale = 2)
    private BigDecimal valorMatricula;

    @Column(name = "valor_salario", precision = 12, scale = 2)
    private BigDecimal valorSalario;
}
